### PR TITLE
infoschema,executor: Add CHECK_CONSTRAINTS I_S table

### LINKED ISSUE #46427: Add `CHECK_CONSTRAINTS` table for mysql8 compatibility
## Feature Request

**Is your feature request related to a problem? Please describe:**

After PR #46155 merged into master, the change in MySQL version number caused the [Prisma's integration tests](https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fmerged_integration_prisma_test/activity) continuously report errors.

The failure reason is because Prisma found that `CHECK_CONSTRAINTS` table does not exist. I guess because Prisma decides whether to read the table based on the version number of mysql.

e8025d8b-c7ad-481d-a590-28058aeae92d

**Describe the feature you'd like:**

Add the `CHECK_CONSTRAINTS` meta-information table before a new version is released, otherwise this will prevent Prisma users from using the latest version of TiDB properly:

https://dev.mysql.com/doc/mysql-infoschema-excerpt/8.0/en/information-schema-check-constraints-table.html

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

- This adds the [`information_schema.CHECK_CONSTRAINTS`  table](https://dev.mysql.com/doc/mysql-infoschema-excerpt/8.1/en/information-schema-check-constraints-table.html) similar to what MySQL 8.0 does.

Note that other databases implement this as well:
```

postgres=# \d information_schema.CHECK_CONSTRAINTS
                       View "information_schema.check_constraints"
       Column       |               Type                | Collation | Nullable | Default 
--------------------+-----------------------------------+-----------+----------+---------
 constraint_catalog | information_schema.sql_identifier |           |          | 
 constraint_schema  | information_schema.sql_identifier |           |          | 
 constraint_name    | information_schema.sql_identifier |           |          | 
 check_clause       | information_schema.character_data |           |          | 

postgres=# SELECT VERSION();
                                                      version                                                      
-------------------------------------------------------------------------------------------------------------------
 PostgreSQL 15.3 (Debian 15.3-0+deb12u1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
(1 row)
```
Issue Number: close #46427 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
The `information_schema.CHECK_CONSTRAINTS` table was added for improved compatibility with MySQL 8.0
```

### CODE REVIEW

**Review by mjonss — APPROVED** [review]

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -627,6 +629,31 @@ func (e *memtableRetriever) setDataFromTables(ctx context.Context, sctx sessionc
...
+func (e *memtableRetriever) setDataFromCheckConstraints(sctx sessionctx.Context, schemas []*model.DBInfo) error {
+	var rows [][]types.Datum
+	checker := privilege.GetPrivilegeManager(sctx)
+	for _, schema := range schemas {
+		for _, table := range schema.Tables {
+			if len(table.Constraints) > 0 {
+				if checker != nil && !checker.RequestVerification(sctx.GetSessionVars().ActiveRoles, schema.Name.L, table.Name.L, "", mysql.SelectPriv) {
+					continue
+				}
+				for _, constraint := range table.Constraints {
```
Comment (fzzf678) [inline executor/infoschema_reader.go:641]: Here `table.Constraints` will return all check constrains in any stats(`StateDeleteOnly`, `StateDeleteReorganization` ...), is this by design or not?

File: infoschema/test/clustertablestest/tables_test.go [hunk 2]
```diff
@@ -1811,3 +1811,16 @@ func TestAddFieldsForBinding(t *testing.T) {
...
 	require.Equal(t, rows[0][8], "select * from `t` where `a` = ?")
 }
+
+func TestCheckConstraints(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("SET GLOBAL tidb_enable_check_constraint = ON")
+	tk.MustExec("CREATE TABLE test.t1 (id INT PRIMARY KEY, CHECK (id<10))")
+	rows := tk.MustQuery("SELECT * FROM information_schema.CHECK_CONSTRAINTS").Rows()
```
Comment (fzzf678) [inline infoschema/test/clustertablestest/tables_test.go:1821]: Could you please add a case about what's the `SELECT * FROM information_schema.CHECK_CONSTRAINTS ` result after dropping a constraint

Comment (dveeden) [reply infoschema/test/clustertablestest/tables_test.go:1821]: Done

**Review by fzzf678 — APPROVED** [review]

**Pull request merged**